opt/scopeserver/dataserver/utils/search.py

```diff
@@ -0,0 +1,267 @@
+            + sorted(good_match, key=sort_matches)
+            + sorted(bad_match, key=sort_matches)
+        )
+
+        return final_matches
+
+    def aggregate_matches(self, matches: List[SearchMatch]) -> Dict[Tuple[str, str], List[str]]:
+        """
+        Collapse matches based on returned element.
+
+        match[0]: Match from search space
+        match[1]: Category of match
+        match[2]: Relationship
+        """
+
+        aggregated_matches: Dict[Tuple[str, str], List[str]] = OrderedDict()
+
+        if self.cross_species == "":
+            for match in matches:
+                key = (match[1], match[2])
+                if key not in aggregated_matches.keys():
+                    aggregated_matches[key] = [match[0]]
+                else:
+                    aggregated_matches[key].append(match[0])
+
+        elif self.cross_species == "hsap":
+            for match in matches:
+                key = (match[1], match[2])
+                for droso_gene in self.dfh.hsap_to_dmel_mappings[match[0]]:
+                    if key not in aggregated_matches.keys():
+                        aggregated_matches[key] = [droso_gene[0]]
+                    else:
+                        aggregated_matches[key].append(droso_gene[0])
+                    self.cross_species_identity[match[0]] = droso_gene[1]
+
+        elif self.cross_species == "mmus":
+            for match in matches:
+                key = (match[1], match[2])
+                for droso_gene in self.dfh.mmus_to_dmel_mappings[self.search_space[match]]:
+                    if key not in aggregated_matches.keys():
+                        aggregated_matches[key] = [droso_gene[0]]
+                    else:
+                        aggregated_matches[key].append(droso_gene[0])
+                    self.cross_species_identity[match[0]] = droso_gene[1]
+
+        return aggregated_matches
+
+    def create_feature_description(
+        self, aggregated_matches: Dict[Tuple[str, str], List[str]], features: Dict[Tuple[str, str], str]
+    ) -> Dict[Tuple[str, str], str]:
+        """
+        Generate descriptions for final results.
+
+        Some matches translate into something other than the term searched for. For example,
+        searching for a gene that is a marker of a cluster, will result in the searched gene
+        being translated into the name of the cluster that it is a marker of.
+        """
+
+        descriptions: Dict[Tuple[str, str], str] = {}
+
+        for k, v in aggregated_matches.items():
+            if self.cross_species != "":
+                species_name = constant.SPECIES_MAP[self.cross_species]["short"]
+                identity_perc = self.cross_species_identity[k[0]]
+                descriptions[k] = f"Orthologue of {k[0]}, {identity_perc:.2f}% identity ({species_name} -> Drosophila)"
+            synonyms = v.copy()
+            try:
+                synonyms.remove(k[0])
+            except ValueError:
+                pass
```

Comment:
```suggestion
            synonyms = [synonym for synonym in v if synonym != k[0]]
```

Reply:
The list comprehension is much slower than the copy + remove:
![image](https://user-images.githubusercontent.com/5079892/83000762-1e338a00-a00b-11ea-9f81-1cd8cc76c5a3.png)

Reply:
Yeah that's to be expected. Is this performance critical? Pick what you like the most if you go with copy+remove consider this instead:
```python
from contextlib import suppress
#...
            with suppress(ValueError):
                synonyms.remove(k[0])
```
instead of
```python
            try:
                synonyms.remove(k[0])
            except ValueError:
                pass
```

Reply:
Nice, I'll stick with that